src/Riok.Mapperly/Descriptors/MappingBodyBuilders/ObjectMemberMappingBodyBuilder.cs

```diff
@@ -36,7 +42,15 @@ public static void BuildMappingBody(IMembersContainerBuilderContext<IMemberAssig
                 // eg. target.A = source.B should be mapped before target.A.Id = source.B.Id
                 foreach (var config in memberConfigs.OrderBy(x => x.Target.Path.Count))
                 {
-                    BuildMemberAssignmentMapping(ctx, config);
+                    if (
+                        GetMemberPaths(ctx, config, out var targetMemberPath, out var innerSourceMemberPath)
+                        && targetMemberPath is not null
+                        && innerSourceMemberPath is not null
```

Comment:
Yes, of course. I actually prefer the `!= null` approach, I used this because I thought it was being used here somewhere. But upon a closer look, it isn't the more frequently used.